src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -197,9 +208,26 @@ void primitive_inst::update_shape() {
 
     _impl_params->memory_deps = memory_deps;
     auto new_layouts = _node.type()->calc_output_layouts(_node, *_impl_params);
-    auto new_layout = new_layouts.empty() ? _node.type()->calc_output_layout(_node, *_impl_params) : new_layouts[0];
+
+    int output_idx = _node.is_type<crop>() ? _impl_params->typed_desc<crop>()->output_idx : 0;
+    auto new_layout = new_layouts.empty() ? _node.type()->calc_output_layout(_node, *_impl_params) : new_layouts[output_idx];
     new_layout.data_padding = padding::max(_node.get_primitive()->output_padding, new_layout.data_padding);
 
+    // W/A for split offsets... :(
+    if (_node.is_type<crop>() && _node.get_dependencies().size() > 1) {
+        InferenceEngine::SizeVector startOffset(_impl_params->input_layouts[0].get_partial_shape().size());
```

Comment:
Is it possible to avoid WA in common code? I think crop::calc_output_layouts may return single layout based on output_idx, so part of this WA won't be needed.
Then you need to deal with input_offsets - can it be calculated in crop_impl::create directly?

Reply:
All output layouts are necessary to calculate input offsets but we can not get all output layouts in crop_impl::create. So I moved WA code to crop_inst::calc_output_layouts instead of crop_impl::create. @yeonbok, @vladimir-paramuzov Could you please review updated codes?

Reply:
For the new update, impl_param  is updated within calc_output_layout using const_cast. 
I think there are pros and cons for both impls, previous one and the current one. 
@vladimir-paramuzov  how do you think having such a const_cast?

Reply:
@vladimir-paramuzov could you please review for updated commit to use const_cast to get input offset in calc_output_layouts?